app/Http/Controllers/BillingController.php

```diff
@@ -662,4 +667,29 @@ private function formatHistoricalUsageData($historicalUsageData)
         return $formattedData;
     }
 
+    private function paginate($items, $perPage, $setDefaultOption = true, $options = [])
+    {
+        if($setDefaultOption){
+            $options = ['path' => request()->url(), 'query' => request()->query()];
+        }
+
+        $requestUrl = $this->cleanUrl(request()->url());
+
+        if (isset($options['path']) && env('NGINX_HOST') . $options['path'] == $requestUrl) {
```

Comment:
This seems a bit weird.. can we not use something like `{$_SERVER['HTTP_HOST']}{$_SERVER['REQUEST_URI']}` ?

Reply:
I could do that but then would have to remove the page off the end.

Here is the output for the three different ways.

`$_SERVER['HTTP_HOST']` = "portal.test"
`$_SERVER['REQUEST_URI']` = "/portal/billing/invoices?page=3"

`$requestUrl = $this->cleanUrl(request()->url());`
`$requestUrl` = "portal.test/portal/billing/invoices"

Reply:
Yeah, it was more grabbing the NGINX_HOST from env seems weird to me. What if someone forked this code and wasn't running it on NGINX anymore? Doesn't it make more sense to just use the basic PHP $_SERVER variables?

Reply:
Makes sense. I've adjusted it in the last commit to pull from  `$_SERVER['HTTP_POST']`